e2e/util/recipe.go

```diff
@@ -0,0 +1,290 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ramendr/ramen/e2e/config"
+	"github.com/ramendr/ramen/e2e/types"
+	recipe "github.com/ramendr/recipe/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	two          = 2
+	three        = 3
+	threeHundred = 300
+)
+
+func CreateRecipe(ctx types.TestContext, recipeSpec *config.Recipe) error {
+	log := ctx.Logger()
+
+	recipeC1 := getBaseRecipe(ctx)
+	ns := ctx.AppNamespace()
+
+	if recipeSpec.CheckHook || recipeSpec.ExecHook {
+		recipeC1.Spec.Hooks = getHooks(ns)
+	}
+
+	recipeC1.Spec.Groups = getGroups(ns)
+	recipeC1.Spec.Workflows = getWorkflows(recipeSpec)
+
+	recipeC2 := recipeC1.DeepCopy()
+
+	err := CreateRecipeOn("c1", ctx, recipeC1)
+	if err != nil {
+		return fmt.Errorf("Failed to create recipe %s in namespace %s on c2 cluster: %v", recipeC1.Name,
+			recipeC1.Namespace, err)
+	}
+
+	err = CreateRecipeOn("c2", ctx, recipeC2)
+	if err != nil {
+		return fmt.Errorf("Failed to create recipe %s in namespace %s on c2 cluster: %v", recipeC1.Name,
+			recipeC1.Namespace, err)
+	}
```

Comment:
Don't use hard coded cluster names here. Use a loop over the cluster you want to add the recipe to.

    env := ctx.Env()
    for _, cluster := range []*types.Cluster{env.C1, env.C2} {
        create recipe on cluster...
    }

Check other places in the code that do stuff on multiple clusters.